R/utilities-observed-data.R

```diff
@@ -16,21 +16,30 @@ readObservedDataFile <- function(fileName,
   forbiddenCharacters <- "\ufeff"
 
   if (extension %in% "csv") {
-    observedData <- read.csv(fileName,
-      header = header,
-      check.names = FALSE,
-      encoding = encoding
-    )
+    observedData <- tryCatch({
+      read.csv(fileName,
+               header = header,
+               check.names = FALSE,
+               encoding = encoding
+      )
+    },
+    error = function(e) {
+      warning(messages$warningCSVNotReadCommaSeparatedAttemptingSemicolon(fileName))
+      return(read.csv2(fileName,
+                       header = header,
+                       check.names = FALSE,
+                       encoding = encoding))
+    })
```

Comment:
please add a test with a csv file which cannot be read by read.csv but is read by read.csv2

Reply:
added